show blockchain log entries on errors

euonymos opened the pull request:
When experimenting with failing tests I guess t would be beneficial to see the log. Alternatively we can add another helper which do the job, leaving `testNoErrors`.

euonymos commented:
Looks like CI runs into OOM or something.

anton-k commented on src/Plutus/Test/Model/Blockchain.hs:
```diff
@@ -390,6 +390,8 @@ data Blockchain = Blockchain
     -- in pretty printers for error logs, user names, script names.
     bchNames :: !BchNames
+  , -- todo not the best place
+    stateDump :: Blockchain -> String
```
No need for this function. It seems to be doing something that user can do

anton-k commented on src/Plutus/Test/Model/Blockchain.hs:
```diff
@@ -843,20 +846,26 @@ waitNSlots n = modify' $ \s -> s {bchCurrentSlot = bchCurrentSlot s + n}
+
+    saveState = do
+      s <- get
+      logInfo $ stateDump s s
```
User can do it if user needs that. No need to have forced log

anton-k commented on src/Plutus/Test/Model/Fork/TxExtra.hs:
```diff
@@ -40,14 +41,15 @@ toExtra = Tx mempty
 data Extra = Extra
   { extra'withdraws      :: [Withdraw]
   , extra'certificates   :: [Certificate]
+  , extra'descr          :: Last String
```
Let's keep `Tx` for things that relate to Cardano. This is test concept to improve readability. We can use `BchNames` for that or add this info to something that is not related to `Tx`. For example to Stats. But `BchNames` is preferable. It's the place to keep human readably synonyms for pretty printing.

anton-k commented:
Thanks for your improvements to the lib so far. I will accept the PR if it pass CI. 
Fixing the lint job is ok. It seems that build fails by out of resources exception on CI-node

anton-k commented:
Also thanks for docs update it's very helpful to keep it up to date.

euonymos commented:
@anton-k I fixed lint tips, and reworked the prettifier, now it looks far better.
![image](https://user-images.githubusercontent.com/103441514/168179090-91db6fc1-4aac-462f-a420-3dc81e85121c.png)

anton-k commented:
Looks great!

anton-k commented:
Maybe it would be cool to have option to show short versions of hashes, like github does for commits
